uk.co.compucorp.civicrm.tasksassignments/tests/phpunit/api/v3/DocumentTest.php

```diff
@@ -304,6 +304,46 @@ public function testDocumentCloningOnlyClonesCurrentRevision() {
     $this->assertEquals(0, $clonedCount['values']);
   }
 
+  public function testOnlyDocumentsAssignedToACaseCreateRevisionsOnUpdate() {
+    $assignment = AssignmentFabricator::fabricate();
+
+    $documents['standAloneDocument'] = DocumentFabricator::fabricateWithAPI();
+    $documents['caseDocument'] = DocumentFabricator::fabricateWithAPI(['case_id' => $assignment->id]);
+    $activityCount = civicrm_api3('Activity', 'getcount');
+    $this->assertEquals(2, $activityCount);
+
+    foreach ($documents as $currentDocument) {
+      DocumentFabricator::fabricateWithAPI([
+        'id' => $currentDocument['id'],
+        'expire_date' => date('Y-m-d'),
+        'remind_me' => 1,
+        'status_id' => CRM_Tasksassignments_BAO_Document::STATUS_APPROVED,
+      ]);
+    }
+
+    $activityCount = civicrm_api3('Activity', 'getcount');
+    $this->assertEquals(3, $activityCount);
+
+    $this->assertDocumentRevisionCount($documents['standAloneDocument']['id'], 1);
```

Comment:
I think it is better to change assertDocumentRevisionCount to getDocumentRevisionCount and do the assertion inside testOnlyDocumentsAssignedToACaseCreateRevisionsOnUpdate itself